service/src/main/kotlin/app/cash/backfila/ui/components/ProgressBar.kt

```diff
@@ -5,16 +5,31 @@ import kotlinx.html.TagConsumer
 import kotlinx.html.div
 import kotlinx.html.style
 
-fun TagConsumer<*>.ProgressBar(numerator: Number, denominator: Number) {
+fun TagConsumer<*>.ProgressBar(
+  numerator: Number,
+  denominator: Number,
+  precomputingDone: Boolean = true,
+) {
   div("w-full bg-gray-200 rounded-full dark:bg-gray-700") {
-    val percentComplete = if (numerator.toInt() == 0) 0 else round((numerator.toDouble() / denominator.toDouble()) * 100)
-    // Don't show blue sliver for 0%, just show the gray empty bar
-    val showPartialBarStyle = if (percentComplete.toInt() != 0) "bg-blue-600" else ""
-    div(
-      "$showPartialBarStyle text-xs font-medium text-blue-100 text-center p-0.5 leading-none rounded-full",
-    ) {
-      style = "width: ${if (percentComplete.toInt() == 0) 100 else percentComplete}%"
-      +"""$percentComplete%"""
+    val percentComplete = when {
+      !precomputingDone -> null // Show indeterminate during precomputing
+      denominator.toDouble() <= 0 -> 0.0 // Handle zero denominator
+      else -> round((numerator.toDouble() / denominator.toDouble()) * 100)
+    }
+
+    // Show indeterminate progress bar during precomputing
+    if (percentComplete == null) {
+      div("animate-pulse bg-gray-300 text-xs font-medium text-center p-0.5 leading-none rounded-full") {
+        style = "width: 100%"
+        +"Computing..."
```

Comment:
```suggestion
        +"..."
```

Not sure we need "computing" in 3 spots in the row since it shows up in the cells beside it too